osclib/freeze_command.py

```diff
@@ -116,8 +116,18 @@ def verify_bootstrap_copy_codes(self, codes):
                     return False
         return True
 
-    def perform(self, prj, copy_bootstrap=True):
+    def perform(self, prj, copy_bootstrap=True, freeze_adi=False):
         self.prj = prj
+
+        if freeze_adi:
+            src_prj = self.api.find_devel_project_from_adi_frozenlinks(self.prj)
+            if src_prj is None:
+                raise Exception("{} does not have a valid frozenlinks".format(self.prj))
+            else:
+                self.api.update_adi_frozenlinks(self.prj, src_prj)
+            # do not continue
```

Comment:
`return` = "do not continue". A comment should indicate what the code does not. In this case two different implementations of freeze and adi implementation is done so do not perform non-adi freeze.